### PR TITLE
Add a text-box component and logtype

### PR DESCRIPTION
**Pull request by jleibs**
I wanted to just include some updating text as part of the view layout and realized that our "log-style" text entries weren't exactly what I wanted. Some of the naming here is probably worth discussing.  In particular, how this should (or shouldn't) line up with the TextEntry logging APIs which serve a very different purpose with a very different view.

Seems like there could be some fun extensions on top of this to do HTML, Markdown, etc. for dynamically showing content along with the data.  Seems particularly relevant with all the LLM / Prompt stuff going on these days.

Example of the motivating use-case:

https://user-images.githubusercontent.com/3312232/235387090-5894b10a-222b-471c-ab6a-980daf7548bf.mp4

This feature is gated behind a new `experimental` module:
```
import rerun.experimental as rr_exp

rr_exp.log_text_box("my_text_box", "Hello world")
```

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2011

### CODE REVIEW

File: crates/re_viewer/src/ui/view_textbox/scene.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use re_log_types::component_types::{self};
+use re_query::{query_entity_with_primary, QueryError};
+
+use crate::{ui::SceneQuery, ViewerContext};
+
+// ---
+
+#[derive(Debug, Clone)]
+pub struct TextboxEntry {
+    pub body: String,
```
Comment (emilk) [inline crates/re_viewer/src/ui/view_textbox/scene.rs:12]:
```suggestion
    pub body: String, // TODO(#1887): avoid allocations
```

Comment (jleibs) [reply crates/re_viewer/src/ui/view_textbox/scene.rs:12]: Ahh, nice. That ticket also suggests:
```
We should also take the time to split TextEntry into two components: text and level.
```

Maybe we should just do that all now.

Comment (emilk) [reply crates/re_viewer/src/ui/view_textbox/scene.rs:12]: It feels like the new view this PR introduces should just be a different view of the same component. The `TextLog` view shows text over time, while the new `TextBox` view shows the latest text(s) at the given time. That would make the difference a difference in blueprint, not in data or component type.

**Comment by emilk** [comment]
> Seems like there could be some fun extensions on top of this to do HTML, Markdown, etc.

A fun use-case for this feature: all our examples could log their own source-code so that when you view one of our examples, you have easy access to the source that created it. If we have a MIME component (or similar) then we could add syntax highlighting to the shown code.

File: crates/re_viewer/src/ui/view_textbox/scene.rs (hunk 1 above)
Comment (jleibs) [reply crates/re_viewer/src/ui/view_textbox/scene.rs:12]: (If only we had a way to choose the view in the blueprint 😅)

Having a marker-component to hint at the right View and different APIs for logging still seems important for user discoverability and ergonomics. These are very different use-cases. Forcing blueprint-interactions to differentiate them doesn’t feel like a great experience.

**Comment by teh-cmc** [comment]
Should we draft this as "blocked by #1887" then?

**Comment by emilk** [comment]
Let's put this behind some sort of experimental namespace and merge this

File: crates/re_viewer/src/ui/view_text_box/ui.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+use super::SceneTextBox;
+
+// --- Main view ---
+
+#[derive(Clone, Default, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct ViewTextBoxState {
+    monospace: bool,
+    word_wrap: bool,
```
Comment (emilk) [inline crates/re_viewer/src/ui/view_text_box/ui.rs:12]: Doesn't it make more sense for word wrapping to be on by default?

**Review by emilk — APPROVED** [review]

**Pull request merged**